Hosted docs

dima618 opened the pull request:
Added modified compodoc configuration to include swagger. Also added compodoc to github pages to host documentation

smalhotra3599 commented:
Lets wait to merge this until we discuss the CI/CD aspect of the auto documentation. While the documentation is great, I don't think we need to commit all 135 files to master. For anything auto generated its typically best practice to keep in the git ignore like node_modules for packages and dist for our build directory. Our CI/CD pipeline can generate the docs at run time without having to commit their files (at least I think)

smalhotra3599 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [CI/CD documentation for backend with github wiki - integrate with swagger/compodoc/postman/other](https://trello.com/c/LvdMcNUx/144-ci-cd-documentation-for-backend-with-github-wiki-integrate-with-swagger-compodoc-postman-other)

dima618 commented:
> Lets wait to merge this until we discuss the CI/CD aspect of the auto documentation. While the documentation is great, I don't think we need to commit all 135 files to master. For anything auto generated its typically best practice to keep in the git ignore like node_modules for packages and dist for our build directory. Our CI/CD pipeline can generate the docs at run time without having to commit their files (at least I think)

Those files need to be committed to the repo so that github pages can host the docs.

smalhotra3599 commented:
> > Lets wait to merge this until we discuss the CI/CD aspect of the auto documentation. While the documentation is great, I don't think we need to commit all 135 files to master. For anything auto generated its typically best practice to keep in the git ignore like node_modules for packages and dist for our build directory. Our CI/CD pipeline can generate the docs at run time without having to commit their files (at least I think)
> 
> Those files need to be committed to the repo so that github pages can host the docs.

Not necessarily (I think), there's a cmd command you can run to publish to github pages. In our CI/CD pipeline we can generate the documentation and use that command to push up and publish every time something gets pushed into master. I attached this link to the trello card as well but I'll also post it here: https://medium.com/front-end-weekly/ci-cd-with-github-actions-to-deploy-on-github-pages-73e225f8f131. This specific example is for building an application and publishing the dist folder after a build but the same logic applies.

dima618 committed:
Create peer-rec-CI.yml

Github action for automatically running unit tests and deploying docs to github pages.

dima618 committed:
Update peer-rec-CI.yml

Removed unnecessary line

dima618 committed:
changed triggers

dima618 committed:
temporarily removed running test command

dima618 committed:
Changed gitignore file to include docs folder

dima618 committed:
Added gh-pages package and commands

dima618 committed:
Merge remote-tracking branch 'Benedict-Scrumberbatch/gh-pages-actions' into hosted-docs

dima618 committed:
changed branch triggers

dima618 committed:
Update repo token

dima618 committed:
Merge branch 'hosted-docs' of https://github.com/benedict-scrumberbatch/peer-recognition-backend into hosted-docs

dima618 committed:
Unnecessary doc files

dima618 committed:
added command to generate docs

dima618 committed:
Merge branch 'hosted-docs' of https://github.com/benedict-scrumberbatch/peer-recognition-backend into hosted-docs

dima618 commented:
I've added the github actions to deploy whenever a pull request is made to master.

smalhotra3599 commented on .github/workflows/peer-rec-CI.yml:
```diff
@@ -0,0 +1,44 @@
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master, hosted-docs ]
+  pull_request:
```
this is great! Lets have this run on every push to master. We don't need the hosted-docs branch necessarily and we don't want to update the live docs until we get an approval. PRs going forward may have comment standards to help with auto documentation generation but you guys can decide on that as a team

smalhotra3599 commented on .github/workflows/peer-rec-CI.yml:
```diff
@@ -0,0 +1,44 @@
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
```
more of a question - are these steps running in sequence or in parallel? Either is fine for now but eventually the building/deployment and documenting scripts should be ran concurrently as part of our CD script on a push to master. Our CI script would run on every pull request and would also run concurrently. The jobs/checks for a pull request include developer approval, passing tests, and a successful build.

dima618 replied:
They're running in sequence. I think we want to generate new docs only if tests/builds pass.

smalhotra3599 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [CI/CD documentation for backend with github wiki - integrate with swagger/compodoc/postman/other](https://trello.com/c/LvdMcNUx/144-ci-cd-documentation-for-backend-with-github-wiki-integrate-with-swagger-compodoc-postman-other)

dima618 committed:
removed hosted-docs from triggers

dima618 committed:
Update and rename peer-rec-CI.yml to peer-rec-CD.yml

smalhotra3599 commented:
do swagger-docs and swagger-assets need to be merged into master or are they autogenerated files? should they be in our .gitignore like docs is

dima618 committed:
Merge branch 'master' into hosted-docs

dima618 committed:
Added separate command for locally hosted docs.
